app/src/main/java/org/p2p/wallet/swap/ui/orca/OrcaSwapPresenter.kt

```diff
@@ -391,9 +391,10 @@ class OrcaSwapPresenter(
             view?.showButtonText(R.string.swap_searching_swap_pair)
             searchTradablePairs(source, destination)
             view?.showButtonText(R.string.swap_calculating_fees)
-            calculateAmount(source, destination)
-            calculateFees(source, destination)
-            calculateRates(source, destination)
+            if (!swapInteractor.getFeePayerToken().isSOL) {
+                swapInteractor.setFeePayerToken(source)
+            }
```

Comment:
because it does not changes any where else but we need to change it when we change source token
FeePayerToken changed only on settings changes
and we have SOL set by default 

that is the issue

Reply:
maybe i can remove check on sol and update it any time
swapInteractor.setFeePayerToken(source) it will work also